From 3d5baf92a1ffb38300dac2fa72751d87b524007c Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Thu, 15 Nov 2018 20:30:36 -0800
Subject: [PATCH] TRITON-941 binder_request_latency_ms should be
 binder_request_latency_seconds Reviewed by: Trent Mick <trentm@gmail.com>
 Approved by: Trent Mick <trentm@gmail.com>

---
 lib/server.js | 4 ++--
 package.json  | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/server.js b/lib/server.js
index 078b4d3..13c764b 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -31,7 +31,7 @@ dtp.enable();
 
 // Binder metric collectors
 var METRIC_REQUEST_COUNTER = 'binder_requests_completed';
-var METRIC_LATENCY_HISTOGRAM = 'binder_request_latency_ms';
+var METRIC_LATENCY_HISTOGRAM = 'binder_request_latency_seconds';
 var METRIC_SIZE_HISTOGRAM = 'binder_response_size_bytes';
 
 ///--- Helpers
@@ -526,7 +526,7 @@ function createServer(options) {
 
                 var labels = { type: query.type() };
                 query_counter.increment(labels);
-                latency_histogram.observe(lat, labels);
+                latency_histogram.observe(lat / 1000, labels);
                 size_histogram.observe(query.bytesSent, labels);
 
                 query._log[loglevel]({
diff --git a/package.json b/package.json
index 8f844b0..1aa8c7f 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "binder",
   "description": "SmartDataService DNS Service",
-  "version": "1.2.1",
+  "version": "1.3.0",
   "author": "Joyent (joyent.com)",
   "license": "MPL-2.0",
   "private": true,
-- 
2.21.0

